core/http/src/parse/uri/tests.rs

```diff
@@ -89,6 +103,25 @@ fn bad_parses() {
     assert_no_parse!("://z7:77777777777777777777777777777`77777777777");
 }
 
+#[test]
+fn test_parse_issue_924_samples() {
+    assert_parses!("/path?param={value}", "/path/?param={value}", "/some/path/?param={forgot-to-replace-placeholder}");
+    assert_parses!("/path?param={value}&onemore={value}");
+    assert_parses!("/some/path/?tags=[]", "/some/path/?tags=[rocket,is,perfect]");
+    assert_parses!("/some/path/?tags=[rocket,is,perfect]&clients={arenot}");
+    assert_parses!("/rocket/@user/");
+    assert_parses!("/rocket/@user/?tags=[rocket,%F0%9F%98%8B]");
+    assert_parses!("/rocket/?username=@sergio&tags=[rocket,%F0%9F%98%8B]");
+    assert_parses!("/rocket/?Key+With+Spaces=value+too");
+    assert_no_parse!("/rocket/?Key+With+\'");
+    assert_parses!("/rocket/?query=%3E5");
+}
+
+// #[test]
```

Comment:
As above, this will probably have to split into "parse arbitrary URI" vs "parse a route specification".